### PR TITLE
Sdks 294 device profile callback

### PR DESCRIPTION
**Pull request by jhoongo**

### CODE REVIEW

File: FRAuth/FRAuth/Callbacks/DeviceProfileCallback.swift [hunk 1]
```diff
@@ -12,12 +12,16 @@
 import Foundation
 
 /// DeviceAttributeCallback is a callback class that collects Device Information using DeviceCollector(s) in FRAuth SDK.
```
Comment (rodrigoareis) [inline FRAuth/FRAuth/Callbacks/DeviceProfileCallback.swift:14]: The class name was updated to DeviceProfileCallback, should it be reflected in the doc?

File: FRAuth/FRAuth/Callbacks/DeviceProfileCallback.swift [hunk 2]
```diff
@@ -26,18 +30,27 @@ import Foundation
...
+            }
+            else if let outputName = output["name"] as? String, outputName == "message", let outputValue = output["value"] as? String {
+                message = outputValue
             }
         }
+        
+        try super.init(json: json)
+        
+        // For now, force Callback.type as DeviceAttributeCallback
+        type = "DeviceAttributeCallback"
```
Comment (rodrigoareis) [inline FRAuth/FRAuth/Callbacks/DeviceProfileCallback.swift:53]: Should the type be updated to DeviceProfileCallback?

File: FRAuth/FRAuth/Callbacks/DeviceProfileCallback.swift [hunk 3]
```diff
@@ -59,28 +72,25 @@ import Foundation
...
-                    FRLog.w("LocationCollector is not found during DeviceAttributeCallback.execute")
-                }
+        if locationRequired {
+            if let locationCollector = locationCollector {
+                collector.collectors.append(locationCollector)
             }
-            else if attribute.lowercased() == "profile" {
-                collector.collectors.append(profileCollector)
+            else {
+                FRLog.w("LocationCollector is not found during DeviceAttributeCallback.execute")
```
Comment (rodrigoareis) [inline FRAuth/FRAuth/Callbacks/DeviceProfileCallback.swift:80]: DeviceProfileCallback?

File: FRAuth/FRAuth/Callbacks/DeviceProfileCallback.swift (hunk 1 above)
Comment (jhoongo) [reply FRAuth/FRAuth/Callbacks/DeviceProfileCallback.swift]: Corrected with the latest commit.

File: FRAuth/FRAuth/Callbacks/DeviceProfileCallback.swift (hunk 2 above)
Comment (jhoongo) [reply FRAuth/FRAuth/Callbacks/DeviceProfileCallback.swift]: Corrected with the latest commit, thanks for pointing this out.

File: FRAuth/FRAuth/Callbacks/DeviceProfileCallback.swift (hunk 3 above)
Comment (jhoongo) [reply FRAuth/FRAuth/Callbacks/DeviceProfileCallback.swift]: Corrected.

**Pull request merged**